base/ntuple.jl

```diff
@@ -95,5 +95,5 @@ end
 function reverse(t::NTuple{N}) where N
     ntuple(Val{N}()) do i
         t[end+1-i]
-    end::NTuple
+    end
```

Comment:
Could you leave this part in? It's just an unconditional type assert, so it shouldn't cause any harm.

Reply:
Let's get back to a known good state and we can readd this if it is shown with the benchmarks the regressed to be ok.